
From: Carl Spalletta <cspalletta@yahoo.com>

Remove a whole bunch of prototypes which declare no-longer-present functions.

Signed-off-by: Andrew Morton <akpm@osdl.org>
---

 25-akpm/arch/i386/kernel/cpu/mtrr/mtrr.h |    1 -
 25-akpm/arch/i386/math-emu/fpu_proto.h   |    1 -
 25-akpm/fs/adfs/adfs.h                   |    1 -
 25-akpm/fs/autofs4/autofs_i.h            |    1 -
 25-akpm/fs/befs/befs.h                   |    1 -
 25-akpm/fs/hfs/hfs_fs.h                  |    3 ---
 25-akpm/fs/hfsplus/hfsplus_fs.h          |    1 -
 25-akpm/fs/jfs/jfs_txnmgr.c              |    1 -
 25-akpm/fs/proc/base.c                   |    1 -
 25-akpm/fs/udf/udfdecl.h                 |    2 --
 25-akpm/fs/xfs/linux-2.6/xfs_fs_subr.h   |    1 -
 25-akpm/fs/xfs/quota/xfs_qm.h            |    1 -
 25-akpm/fs/xfs/xfs_acl.h                 |    2 --
 25-akpm/fs/xfs/xfs_attr_leaf.h           |    1 -
 25-akpm/fs/xfs/xfs_bmap_btree.h          |    7 -------
 25-akpm/fs/xfs/xfs_inode.h               |    1 -
 25-akpm/fs/xfs/xfs_log_priv.h            |    1 -
 25-akpm/net/atm/common.h                 |   11 -----------
 25-akpm/net/atm/lec.h                    |    1 -
 25-akpm/net/sctp/socket.c                |    1 -
 25-akpm/security/selinux/include/avc.h   |    1 -
 25-akpm/security/selinux/ss/policydb.h   |    1 -
 22 files changed, 42 deletions(-)

diff -puN arch/i386/kernel/cpu/mtrr/mtrr.h~remove-dead-prototypes arch/i386/kernel/cpu/mtrr/mtrr.h
--- 25/arch/i386/kernel/cpu/mtrr/mtrr.h~remove-dead-prototypes	2004-08-05 11:28:09.129386136 -0700
+++ 25-akpm/arch/i386/kernel/cpu/mtrr/mtrr.h	2004-08-05 11:28:09.161381272 -0700
@@ -52,7 +52,6 @@ struct mtrr_ops {
 };
 
 extern int generic_get_free_region(unsigned long base, unsigned long size);
-extern void generic_init_secondary(void);
 extern int generic_validate_add_page(unsigned long base, unsigned long size,
 				     unsigned int type);
 
diff -puN arch/i386/math-emu/fpu_proto.h~remove-dead-prototypes arch/i386/math-emu/fpu_proto.h
--- 25/arch/i386/math-emu/fpu_proto.h~remove-dead-prototypes	2004-08-05 11:28:09.131385832 -0700
+++ 25-akpm/arch/i386/math-emu/fpu_proto.h	2004-08-05 11:28:09.162381120 -0700
@@ -69,7 +69,6 @@ extern int isNaN(FPU_REG const *ptr);
 extern void FPU_pop(void);
 extern int FPU_empty_i(int stnr);
 extern int FPU_stackoverflow(FPU_REG **st_new_ptr);
-extern void FPU_sync_tags(void);
 extern void FPU_copy_to_regi(FPU_REG const *r, u_char tag, int stnr);
 extern void FPU_copy_to_reg1(FPU_REG const *r, u_char tag);
 extern void FPU_copy_to_reg0(FPU_REG const *r, u_char tag);
diff -puN fs/adfs/adfs.h~remove-dead-prototypes fs/adfs/adfs.h
--- 25/fs/adfs/adfs.h~remove-dead-prototypes	2004-08-05 11:28:09.132385680 -0700
+++ 25-akpm/fs/adfs/adfs.h	2004-08-05 11:28:09.162381120 -0700
@@ -71,7 +71,6 @@ struct adfs_discmap {
 int adfs_get_block(struct inode *inode, sector_t block,
 		   struct buffer_head *bh, int create);
 struct inode *adfs_iget(struct super_block *sb, struct object_info *obj);
-void adfs_read_inode(struct inode *inode);
 void adfs_write_inode(struct inode *inode,int unused);
 int adfs_notify_change(struct dentry *dentry, struct iattr *attr);
 
diff -puN fs/autofs4/autofs_i.h~remove-dead-prototypes fs/autofs4/autofs_i.h
--- 25/fs/autofs4/autofs_i.h~remove-dead-prototypes	2004-08-05 11:28:09.133385528 -0700
+++ 25-akpm/fs/autofs4/autofs_i.h	2004-08-05 11:28:09.162381120 -0700
@@ -138,7 +138,6 @@ static inline void autofs4_copy_atime(st
 }
 
 struct inode *autofs4_get_inode(struct super_block *, struct autofs_info *);
-struct autofs_info *autofs4_init_inf(struct autofs_sb_info *, mode_t mode);
 void autofs4_free_ino(struct autofs_info *);
 
 /* Expiration */
diff -puN fs/befs/befs.h~remove-dead-prototypes fs/befs/befs.h
--- 25/fs/befs/befs.h~remove-dead-prototypes	2004-08-05 11:28:09.135385224 -0700
+++ 25-akpm/fs/befs/befs.h	2004-08-05 11:28:09.163380968 -0700
@@ -96,7 +96,6 @@ void befs_dump_super_block(const struct 
 void befs_dump_inode(const struct super_block *sb, befs_inode *);
 void befs_dump_index_entry(const struct super_block *sb, befs_btree_super *);
 void befs_dump_index_node(const struct super_block *sb, befs_btree_nodehead *);
-void befs_dump_inode_addr(const struct super_block *sb, befs_inode_addr);
 /****************************/
 
 
diff -puN fs/hfs/hfs_fs.h~remove-dead-prototypes fs/hfs/hfs_fs.h
--- 25/fs/hfs/hfs_fs.h~remove-dead-prototypes	2004-08-05 11:28:09.136385072 -0700
+++ 25-akpm/fs/hfs/hfs_fs.h	2004-08-05 11:28:09.163380968 -0700
@@ -223,9 +223,6 @@ extern int hfs_strcmp(const unsigned cha
 		      const unsigned char *, unsigned int);
 extern int hfs_compare_dentry(struct dentry *, struct qstr *, struct qstr *);
 
-/* super.c */
-extern struct super_block *hfs_read_super(struct super_block *,void *,int);
-
 /* trans.c */
 extern void hfs_triv2mac(struct hfs_name *, struct qstr *);
 extern int hfs_mac2triv(char *, const struct hfs_name *);
diff -puN fs/hfsplus/hfsplus_fs.h~remove-dead-prototypes fs/hfsplus/hfsplus_fs.h
--- 25/fs/hfsplus/hfsplus_fs.h~remove-dead-prototypes	2004-08-05 11:28:09.137384920 -0700
+++ 25-akpm/fs/hfsplus/hfsplus_fs.h	2004-08-05 11:28:09.164380816 -0700
@@ -344,7 +344,6 @@ int hfsplus_ioctl(struct inode *inode, s
 /* options.c */
 int parse_options(char *, struct hfsplus_sb_info *);
 void fill_defaults(struct hfsplus_sb_info *);
-void fill_current(struct hfsplus_sb_info *, struct hfsplus_sb_info *);
 
 /* tables.c */
 extern u16 case_fold_table[];
diff -puN fs/jfs/jfs_txnmgr.c~remove-dead-prototypes fs/jfs/jfs_txnmgr.c
--- 25/fs/jfs/jfs_txnmgr.c~remove-dead-prototypes	2004-08-05 11:28:09.139384616 -0700
+++ 25-akpm/fs/jfs/jfs_txnmgr.c	2004-08-05 11:28:09.166380512 -0700
@@ -165,7 +165,6 @@ struct {
  * external references
  */
 extern int lmGroupCommit(struct jfs_log *, struct tblock *);
-extern void lmSync(struct jfs_log *);
 extern int jfs_commit_inode(struct inode *, int);
 extern int jfs_stop_threads;
 
diff -puN fs/proc/base.c~remove-dead-prototypes fs/proc/base.c
--- 25/fs/proc/base.c~remove-dead-prototypes	2004-08-05 11:28:09.140384464 -0700
+++ 25-akpm/fs/proc/base.c	2004-08-05 11:28:09.167380360 -0700
@@ -192,7 +192,6 @@ static inline int proc_type(struct inode
 int proc_pid_stat(struct task_struct*,char*);
 int proc_pid_status(struct task_struct*,char*);
 int proc_pid_statm(struct task_struct*,char*);
-int proc_pid_cpu(struct task_struct*,char*);
 
 static int proc_fd_link(struct inode *inode, struct dentry **dentry, struct vfsmount **mnt)
 {
diff -puN fs/udf/udfdecl.h~remove-dead-prototypes fs/udf/udfdecl.h
--- 25/fs/udf/udfdecl.h~remove-dead-prototypes	2004-08-05 11:28:09.142384160 -0700
+++ 25-akpm/fs/udf/udfdecl.h	2004-08-05 11:28:09.168380208 -0700
@@ -111,7 +111,6 @@ extern int8_t udf_next_aext(struct inode
 extern int8_t udf_current_aext(struct inode *, lb_addr *, int *, lb_addr *, uint32_t *, struct buffer_head **, int);
 
 /* misc.c */
-extern int udf_read_tagged_data(char *, int size, int fd, int block, int partref);
 extern struct buffer_head *udf_tgetblk(struct super_block *, int);
 extern struct buffer_head *udf_tread(struct super_block *, int);
 extern struct genericFormat *udf_add_extendedattr(struct inode *, uint32_t, uint32_t, uint8_t);
@@ -163,7 +162,6 @@ extern struct fileIdentDesc * udf_get_fi
 extern extent_ad * udf_get_fileextent(void * buffer, int bufsize, int * offset);
 extern long_ad * udf_get_filelongad(uint8_t *, int, int *, int);
 extern short_ad * udf_get_fileshortad(uint8_t *, int, int *, int);
-extern uint8_t * udf_get_filead(struct fileEntry *, uint8_t *, int, int, int, int *);
 
 /* crc.c */
 extern uint16_t udf_crc(uint8_t *, uint32_t, uint16_t);
diff -puN fs/xfs/linux-2.6/xfs_fs_subr.h~remove-dead-prototypes fs/xfs/linux-2.6/xfs_fs_subr.h
--- 25/fs/xfs/linux-2.6/xfs_fs_subr.h~remove-dead-prototypes	2004-08-05 11:28:09.143384008 -0700
+++ 25-akpm/fs/xfs/linux-2.6/xfs_fs_subr.h	2004-08-05 11:28:09.168380208 -0700
@@ -40,7 +40,6 @@ struct cred;
 
 extern int	fs_noerr(void);
 extern int	fs_nosys(void);
-extern int	fs_nodev(void);
 extern void	fs_noval(void);
 extern void	fs_tosspages(bhv_desc_t *, xfs_off_t, xfs_off_t, int);
 extern void	fs_flushinval_pages(bhv_desc_t *, xfs_off_t, xfs_off_t, int);
diff -puN fs/xfs/quota/xfs_qm.h~remove-dead-prototypes fs/xfs/quota/xfs_qm.h
--- 25/fs/xfs/quota/xfs_qm.h~remove-dead-prototypes	2004-08-05 11:28:09.144383856 -0700
+++ 25-akpm/fs/xfs/quota/xfs_qm.h	2004-08-05 11:28:09.168380208 -0700
@@ -187,7 +187,6 @@ extern int		xfs_qm_write_sb_changes(xfs_
 extern int		xfs_qm_sync(xfs_mount_t *, short);
 
 /* dquot stuff */
-extern void		xfs_qm_dqunlink(xfs_dquot_t *);
 extern boolean_t	xfs_qm_dqalloc_incore(xfs_dquot_t **);
 extern int		xfs_qm_dqattach(xfs_inode_t *, uint);
 extern void		xfs_qm_dqdetach(xfs_inode_t *);
diff -puN fs/xfs/xfs_acl.h~remove-dead-prototypes fs/xfs/xfs_acl.h
--- 25/fs/xfs/xfs_acl.h~remove-dead-prototypes	2004-08-05 11:28:09.146383552 -0700
+++ 25-akpm/fs/xfs/xfs_acl.h	2004-08-05 11:28:09.169380056 -0700
@@ -74,8 +74,6 @@ extern struct kmem_zone *xfs_acl_zone;
 
 extern int xfs_acl_inherit(struct vnode *, struct vattr *, xfs_acl_t *);
 extern int xfs_acl_iaccess(struct xfs_inode *, mode_t, cred_t *);
-extern int xfs_acl_get(struct vnode *, xfs_acl_t *, xfs_acl_t *);
-extern int xfs_acl_set(struct vnode *, xfs_acl_t *, xfs_acl_t *);
 extern int xfs_acl_vtoacl(struct vnode *, xfs_acl_t *, xfs_acl_t *);
 extern int xfs_acl_vhasacl_access(struct vnode *);
 extern int xfs_acl_vhasacl_default(struct vnode *);
diff -puN fs/xfs/xfs_attr_leaf.h~remove-dead-prototypes fs/xfs/xfs_attr_leaf.h
--- 25/fs/xfs/xfs_attr_leaf.h~remove-dead-prototypes	2004-08-05 11:28:09.147383400 -0700
+++ 25-akpm/fs/xfs/xfs_attr_leaf.h	2004-08-05 11:28:09.169380056 -0700
@@ -246,7 +246,6 @@ int	xfs_attr_shortform_getvalue(struct x
 int	xfs_attr_shortform_to_leaf(struct xfs_da_args *args);
 int	xfs_attr_shortform_remove(struct xfs_da_args *remove);
 int	xfs_attr_shortform_list(struct xfs_attr_list_context *context);
-int	xfs_attr_shortform_replace(struct xfs_da_args *args);
 int	xfs_attr_shortform_allfit(struct xfs_dabuf *bp, struct xfs_inode *dp);
 
 /*
diff -puN fs/xfs/xfs_bmap_btree.h~remove-dead-prototypes fs/xfs/xfs_bmap_btree.h
--- 25/fs/xfs/xfs_bmap_btree.h~remove-dead-prototypes	2004-08-05 11:28:09.148383248 -0700
+++ 25-akpm/fs/xfs/xfs_bmap_btree.h	2004-08-05 11:28:09.170379904 -0700
@@ -551,13 +551,6 @@ xfs_bmbt_insert(
 	struct xfs_btree_cur *,
 	int *);
 
-int
-xfs_bmbt_insert_many(
-	struct xfs_btree_cur *,
-	int,
-	xfs_bmbt_rec_t *,
-	int *);
-
 void
 xfs_bmbt_log_block(
 	struct xfs_btree_cur *,
diff -puN fs/xfs/xfs_inode.h~remove-dead-prototypes fs/xfs/xfs_inode.h
--- 25/fs/xfs/xfs_inode.h~remove-dead-prototypes	2004-08-05 11:28:09.150382944 -0700
+++ 25-akpm/fs/xfs/xfs_inode.h	2004-08-05 11:28:09.171379752 -0700
@@ -508,7 +508,6 @@ void		xfs_xlate_dinode_core(xfs_caddr_t,
 uint		xfs_dic2xflags(struct xfs_dinode_core *, xfs_arch_t);
 int		xfs_ifree(struct xfs_trans *, xfs_inode_t *,
 			   struct xfs_bmap_free *);
-int		xfs_atruncate_start(xfs_inode_t *);
 void		xfs_itruncate_start(xfs_inode_t *, uint, xfs_fsize_t);
 int		xfs_itruncate_finish(struct xfs_trans **, xfs_inode_t *,
 				     xfs_fsize_t, int, int);
diff -puN fs/xfs/xfs_log_priv.h~remove-dead-prototypes fs/xfs/xfs_log_priv.h
--- 25/fs/xfs/xfs_log_priv.h~remove-dead-prototypes	2004-08-05 11:28:09.151382792 -0700
+++ 25-akpm/fs/xfs/xfs_log_priv.h	2004-08-05 11:28:09.171379752 -0700
@@ -543,7 +543,6 @@ extern int	 xlog_find_tail(xlog_t	*log,
 				xfs_daddr_t *head_blk,
 				xfs_daddr_t *tail_blk,
 				int readonly);
-extern int	 xlog_print_find_oldest(xlog_t *log, xfs_daddr_t *last_blk);
 extern int	 xlog_recover(xlog_t *log, int readonly);
 extern int	 xlog_recover_finish(xlog_t *log, int mfsi_flags);
 extern void	 xlog_pack_data(xlog_t *log, xlog_in_core_t *iclog);
diff -puN net/atm/common.h~remove-dead-prototypes net/atm/common.h
--- 25/net/atm/common.h~remove-dead-prototypes	2004-08-05 11:28:09.152382640 -0700
+++ 25-akpm/net/atm/common.h	2004-08-05 11:28:09.172379600 -0700
@@ -24,11 +24,6 @@ int vcc_setsockopt(struct socket *sock, 
 int vcc_getsockopt(struct socket *sock, int level, int optname,
 		   char __user *optval, int __user *optlen);
 
-void atm_shutdown_dev(struct atm_dev *dev);
-
-void pppoatm_ioctl_set(int (*hook)(struct atm_vcc *, unsigned int, unsigned long));
-void br2684_ioctl_set(int (*hook)(struct atm_vcc *, unsigned int, unsigned long));
-
 int atmpvc_init(void);
 void atmpvc_exit(void);
 int atmsvc_init(void);
@@ -50,12 +45,6 @@ static inline void atm_proc_exit(void)
 #endif /* CONFIG_PROC_FS */
 
 /* SVC */
-
-void svc_callback(struct atm_vcc *vcc);
 int svc_change_qos(struct atm_vcc *vcc,struct atm_qos *qos);
 
-/* p2mp */
-
-int create_leaf(struct socket *leaf,struct socket *session);
-
 #endif
diff -puN net/atm/lec.h~remove-dead-prototypes net/atm/lec.h
--- 25/net/atm/lec.h~remove-dead-prototypes	2004-08-05 11:28:09.154382336 -0700
+++ 25-akpm/net/atm/lec.h	2004-08-05 11:28:09.172379600 -0700
@@ -151,7 +151,6 @@ int lecd_attach(struct atm_vcc *vcc, int
 int lec_vcc_attach(struct atm_vcc *vcc, void __user *arg);
 int lec_mcast_attach(struct atm_vcc *vcc, int arg);
 struct net_device *get_dev_lec(int itf);
-int make_lec(struct atm_vcc *vcc);
 int send_to_lecd(struct lec_priv *priv,
                  atmlec_msg_type type, unsigned char *mac_addr,
                  unsigned char *atm_addr, struct sk_buff *data);
diff -puN net/sctp/socket.c~remove-dead-prototypes net/sctp/socket.c
--- 25/net/sctp/socket.c~remove-dead-prototypes	2004-08-05 11:28:09.156382032 -0700
+++ 25-akpm/net/sctp/socket.c	2004-08-05 11:28:09.175379144 -0700
@@ -108,7 +108,6 @@ static void sctp_sock_migrate(struct soc
 static char *sctp_hmac_alg = SCTP_COOKIE_HMAC_ALG;
 
 extern kmem_cache_t *sctp_bucket_cachep;
-extern int sctp_assoc_valid(struct sock *sk, struct sctp_association *asoc);
 
 /* Get the sndbuf space available at the time on the association.  */
 static inline int sctp_wspace(struct sctp_association *asoc)
diff -puN security/selinux/include/avc.h~remove-dead-prototypes security/selinux/include/avc.h
--- 25/security/selinux/include/avc.h~remove-dead-prototypes	2004-08-05 11:28:09.157381880 -0700
+++ 25-akpm/security/selinux/include/avc.h	2004-08-05 11:28:09.176378992 -0700
@@ -111,7 +111,6 @@ struct avc_audit_data {
 struct audit_buffer;
 void avc_dump_av(struct audit_buffer *ab, u16 tclass, u32 av);
 void avc_dump_query(struct audit_buffer *ab, u32 ssid, u32 tsid, u16 tclass);
-void avc_dump_cache(struct audit_buffer *ab, char *tag);
 
 /*
  * AVC operations
diff -puN security/selinux/ss/policydb.h~remove-dead-prototypes security/selinux/ss/policydb.h
--- 25/security/selinux/ss/policydb.h~remove-dead-prototypes	2004-08-05 11:28:09.158381728 -0700
+++ 25-akpm/security/selinux/ss/policydb.h	2004-08-05 11:28:09.176378992 -0700
@@ -251,7 +251,6 @@ struct policydb {
 extern int policydb_init(struct policydb *p);
 extern int policydb_index_classes(struct policydb *p);
 extern int policydb_index_others(struct policydb *p);
-extern int constraint_expr_destroy(struct constraint_expr *expr);
 extern void policydb_destroy(struct policydb *p);
 extern int policydb_load_isids(struct policydb *p, struct sidtab *s);
 extern int policydb_context_isvalid(struct policydb *p, struct context *c);
_
